An option to enable / disable map '/' with healthcheck response

xiazek opened the pull request:
## Description
<!--- Describe your changes in detail -->
 - allow passing `health_on_root_path`, to be able to skip healthcheck response on `map '/'` (while keeping gem's `map 'health'` and `map '/healthz'`)
 - backward compatible (taking `health_route` as default value )

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Allow to skip or overwrite `map '/'`, while still keeping `/health` without too much custom settings.

knapo reviewed (changes requested):
I think that having a root path setup by default is nice, as most of GQL services doesn't implements any functionality on root path. Also, such change breaks current implementation pretty significantly and except from users to on gem upgrade. 

I'd make it more backwards compatible and only add an option to skip root path setup, so if someone doesn't want it or want to set it up on his/her own 
e.g. `health_on_root_path` option which is `true` by default.

xiazek commented:
> I think that having a root path setup by default is nice, as most of GQL services doesn't implements any functionality on root path. Also, such change breaks current implementation pretty significantly and except from users to on gem upgrade.
> 

hmm... the default argument value `root_path_response_builder: RackGraphql::HealthResponseBuilder,` makes it working exactly the same, without any changes required after update. What does it break in your opinion ?

Any behaviour would change only after passing `root_path_response_builder`.

> I'd make it more backwards compatible and only add an option to skip root path setup, so if someone doesn't want it or want to set it up on his/her own
> e.g. `health_on_root_path` option which is `true` by default.

Such option is fine for me too. I'll push an update later. 👍

knapo commented:
> hmm... the default argument value root_path_response_builder: RackGraphql::HealthResponseBuilder, makes it working exactly the same, without any changes required after update. What does it break in your opinion ?

huh, my bad, thought it's nil by default 🤦

xiazek commented:
> > hmm... the default argument value root_path_response_builder: RackGraphql::HealthResponseBuilder, makes it working exactly the same, without any changes required after update. What does it break in your opinion ?
> 
> huh, my bad, thought it's nil by default 🤦

No problem. Anyway. I like the version:

> [..] e.g. health_on_root_path option which is true by default.

I'll push an update later.

xiazek commented:
Updated and added example to README:

https://github.com/RenoFi/rack-graphql/pull/109/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R32
